core/frontend/src/quantity-formatting/QuantityFormatter.ts

```diff
@@ -1051,8 +1107,41 @@ export class QuantityFormatter implements UnitsProvider {
     const format = await Format.createFromJSON(formatName ?? "temp", this._unitsProvider, formatProps);
     return ParserSpec.create(format, this._unitsProvider, persistenceUnitProps);
   }
-}
 
+  /**
+   * @beta
+   * Returns a [[FormattingSpecEntry]] for a given name, typically a KindOfQuantity full name.
+   */
+  public getSpecsByName(name: string): FormattingSpecEntry | undefined {
+    return this._formatSpecsRegistry.get(name);
+  }
+
+    /**
+   * Populates the registry with a new FormatterSpec and ParserSpec entry for the given format name.
+   * @beta
+   * @param name The key used to identify the formatter and parser spec
+   * @param persistenceUnitName The name of the persistence unit
+   * @param formatProps If not supplied, tries to retrieve the [[FormatProps]] from [[IModelApp.formatsProvider]]
+   */
+  public async addFormattingSpecsToRegistry(name: string, persistenceUnitName: string, formatProps?: FormatProps): Promise<void> {
+    if (!formatProps) {
+      formatProps = await IModelApp.formatsProvider.getFormat(name);
+    }
+    if (formatProps) {
```

Comment:
do we want it to silently fail? maybe it makes more sense to throw an error when we fail to find format props?